torchvision/utils.py

```diff
@@ -210,3 +210,59 @@ def draw_bounding_boxes(
             draw.text((bbox[0], bbox[1]), labels[i], fill=color, font=txt_font)
 
     return torch.from_numpy(np.array(img_to_draw)).permute(2, 0, 1)
+
+
+@torch.no_grad()
+def draw_segmentation_masks(
+    image: torch.Tensor,
+    masks: torch.Tensor,
+    colors: Optional[List[Union[str, Tuple[int, int, int]]]] = None,
+) -> torch.Tensor:
+
+    """
+    Draws segmentation masks on given image.
+    The values of the input image should be uint8 between 0 and 255.
+
+    Args:
+        image (Tensor): Tensor of shape (C x H x W)
+        masks (Tensor): Tensor of shape (H, W). Each containing predicted class.
+        labels (List[str]): List containing the labels of masks.
+        colors (List[Union[str, Tuple[int, int, int]]]): List containing the colors of masks. The colors can
+            be represented as `str` or `Tuple[int, int, int]`.
+    """
+
+    if not isinstance(image, torch.Tensor):
+        raise TypeError(f"Tensor expected, got {type(image)}")
+    elif image.dtype != torch.uint8:
+        raise ValueError(f"Tensor uint8 expected, got {image.dtype}")
+    elif image.dim() != 3:
+        raise ValueError("Pass individual images, not batches")
+
+    img_to_draw = Image.fromarray(masks.byte().cpu().numpy()).resize(image.size()[1:])
+
+    if colors is None:
+        # Default color palette assumes 21 classes.
+        palette = torch.tensor([2 ** 25 - 1, 2 ** 15 - 1, 2 ** 21 - 1])
+        colors_t = torch.as_tensor([i for i in range(21)])[:, None] * palette
+        color_arr = (colors_t % 255).numpy().astype("uint8")
+
+    else:
+        color_list = []
+        for color in colors:
+            if isinstance(color, str):
+                # This will automatically raise Error if rgb cannot be parsed.
+                fill_color = ImageColor.getrgb(color)  # + (100,)
+                color_list.append(fill_color)
+            elif isinstance(color, tuple):
+                # fill_color = color + (100,)
+                # Use the given colors list and create ndarray of colors.
+                color_list.append(color)
+
+        color_arr = np.array(color_list).astype("uint8")
+
+    img_to_draw.putpalette(color_arr)
+    img_to_draw = torch.from_numpy(np.array(img_to_draw))
+
+    # Project the drawn image to orignal one
+    image[: 1] = img_to_draw
```

Comment:
I need small help here as this projects to a black background.

My guess is we need an alpha channel which will make masks transparent ?

Reply:
I supposed there are many ways to do it. One that I had in mind originally (which might not be the optimal) is to convert the `img_to_draw ` from palette to RGBA, replace the background colour with transparent and then combine it with `image` to achieve the "projection". Worth experimenting with the approach because it's likely there is a better way.

Reply:
Yep, DETR has very nice visualization, but they use matplotlib. Unsure how to reproduce them though.
As you pointed out before [Mask RCNN utils](https://github.com/matterport/Mask_RCNN/blob/3deaec5d902d16e1daf56b62d5971d428dc920bc/mrcnn/visualize.py#L72) have nice way to apply mask too.

Reply:
I just sent it for reference not necessarily for reproduction. :)